[Jizhuo] iP

JizhuoChen opened the pull request:
### DukePro

> “Your mind is for having ideas, not holding them.” – David Allen  ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```javascript
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

JizhuoChen committed:
level-0

JizhuoChen committed:
level-1

JizhuoChen committed:
level-2

JizhuoChen committed:
level-3

JizhuoChen committed:
level-4

JizhuoChen committed:
level-5

JizhuoChen committed:
a-textuitesting

JizhuoChen committed:
level-6

JizhuoChen committed:
no message

JizhuoChen committed:
level-7 save

JizhuoChen committed:
Merge branch 'branch-Level-7'

JizhuoChen committed:
level-8 datetime

JizhuoChen committed:
Merge branch 'branch-Level-8'

JizhuoChen committed:
A-MoreOOP

JizhuoChen committed:
Merge branch 'branch-A-MoreOOP'

JizhuoChen committed:
put all classes into a package named duke

JizhuoChen committed:
Merge branch 'add-gradle-support'

JizhuoChen committed:
A-gradle setups

JizhuoChen committed:
added junit tests for 3 classes

JizhuoChen committed:
Merge branch 'branch-A-JUnit'

JizhuoChen committed:
create jar file

JizhuoChen committed:
Add javadoc for 8 classes

JizhuoChen committed:
Tweak the code to comply with the given coding standard

JizhuoChen committed:
Add the findTask function allowing find task using keywords

JizhuoChen committed:
Merge branch 'branch-A-JavaDoc'

JizhuoChen committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/DateTime.java
#	src/main/java/duke/Deadlines.java
#	src/main/java/duke/Events.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/ToDos.java

JizhuoChen committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Deadlines.java
#	src/main/java/duke/Events.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java

AriellaCallista commented on src/main/java/duke/Addlist.java:
```diff
@@ -0,0 +1,29 @@
+
+import java.util.Scanner;
+
+public class Addlist {
```
Class naming should use PascalCase, e.g. `AddList` instead of `Addlist`

AriellaCallista commented on src/main/java/duke/Addlist.java:
```diff
@@ -0,0 +1,29 @@
+    private int counter = 0;
+    private String[] storage = new String[100];
+
+    public void addlist() {
```
Method names should use camelCase, e.g. `addList` instead of `addlist`

AriellaCallista replied:
Perhaps you could make your method name more intuitive? e.g. `addToList` instead of `addList` as this could mean you're adding a list instead of adding **to** a list!

AriellaCallista commented on src/main/java/duke/DateTime.java:
```diff
@@ -0,0 +1,75 @@
+     * @throws DateTimeParseException If the input string cannot be parsed.
+     */
+    public DateTime(String dt) throws DateTimeParseException {
+        String[] splt = dt.split(" ");
```
A very minor suggestion but I would recommend using standard words i.e. `split` instead of `splt`

AriellaCallista commented on src/main/java/duke/Keywords.java:
```diff
@@ -0,0 +1,5 @@
+package duke;
+
+public enum Keywords {
+
```
If you're not using enums in your code, it's best to discard this code/file.

AriellaCallista commented on src/main/java/duke/Scripts.java:
```diff
@@ -0,0 +1,17 @@
+    String greeting = "Hello! I'm LimBay.\nWhat can I do for you?\n";
+    String bye = "Bye. Hope to see you again soon!";
+
+    public Scripts() {
```
Should this have been deleted? Or did you intend to declare the constructor without any implementation? Because this code is a bit redundant hence it's best to discard it

AriellaCallista commented on src/test/java/duke/DateTimeTest.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    @Test
+    public void testConstructor_WithInvalidFormat_ShouldThrowException() {
```
I like how descriptive your test method name is

AriellaCallista replied:
I also like how you tested for erroneous inputs as well

ScarletBlanks commented on src/main/java/Duke.java:
```diff
@@ -1,4 +1,4 @@
-import duke.Taskmanager;
+import duke.TaskMaster;
```
i think this isnt necessary is you have them packaged

ScarletBlanks commented on src/main/java/duke/Storage.java:
```diff
@@ -7,6 +7,8 @@
+import java.util.ArrayList;
+
 /**
  * Represents a storage handler for managing reading from and writing to a file.
```
Was kinda confused by phrasing, instead of "this class" i think you meant "this method"?

ScarletBlanks commented on src/main/java/duke/TaskList.java:
```diff
@@ -74,18 +74,45 @@ public void removeTask(int index) {
+    }
+
+    /**
+     * Finds a task from the ArrayList based on a key.
```
ok but i think it can be phrased better, i had to read the function to figure out what you meant by key, maybe call it a substring?

ScarletBlanks reviewed (commented):
Code standard is overall pretty clean. Some things can be better phrased but overall very nice.

JizhuoChen committed:
Add GUI for duke which follows the JavaFX tutorial

JizhuoChen committed:
Use checkstyle to check if the code follows coding standard

JizhuoChen committed:
Add assertions to some classes

In classes Todos, Deadlines and Events, assertions are added to document assumptions like descriptions, date or time should not be nul.

In other classes, assertions are added to document some important assumptions that should hold at various points in the code. An example is in Duke.class, user input now cannot be null when clicking the send button.

Using assert statements wisely can lead to more maintainable and robust code. However, as with all tools, it's important to use them judiciously and in the appropriate context.

JizhuoChen committed:
Merge pull request #1 from JizhuoChen/branch-A-Assertions

merge branch-A-Assertions to master branch

JizhuoChen committed:
Change some parts of the code to improve code quality

JizhuoChen committed:
Merge branch 'master' into branch-A-CodeQuality

JizhuoChen committed:
Merge branch 'branch-A-CodeQuality'

JizhuoChen committed:
Add an extension: D-Notes

Add a new feature which allows user to add a note for any task in the list. Users can also peek the notes for any task in the list by providing its index.

More specifically, sending "peekN" + index will show the notes for that task of the index, while sending "editN" + index + notes will update the note for the task to be the supplied note.

This feature is useful in the case when the user forget to include any details in the description of the task, and wish to add a notes to it.

This feature is also relevant to our TP as we also need a feature that allows users to add a bio/short intro of him/herself, the Notes feature can be easily modified to achieve the purpose.

JizhuoChen committed:
Refine the code and add a user-guide

JizhuoChen committed:
Refine the code and put all classes in the duke package
